src/common/Gateway.sol

```diff
@@ -107,8 +105,8 @@ contract Gateway is Auth, IGateway, Recoverable {
 
     /// @inheritdoc IGateway
     function file(bytes32 what, address instance) external auth {
-        if (what == "gasService") gasService = IGasService(instance);
-        else if (what == "processor") processor = IMessageProcessor(instance);
+        if (what == "processor") processor = IMessageProcessor(instance);
+        else if (what == "gasLimit") gasLimit = uint64(uint160(bytes20(instance)));
```

Comment:
`gasLimit` would still be different for `quorum=1` case, which could be the case for some chain IDs but not for others

Reply:
In which sense? `handle()` just stores the message independently of what it is.

Reply:
If `quorum=1`, then the adapter executing `handle()` will pay for the full gas limit of the message processing.

Reply:
Correct, I forgot about the shortcut we have for quorum == 1.

I think we should remove that shortcut 🤔. The problem is that if not, we have two different ways of handling the entire process in different chains:

1. GasLimit computed in chainA for each message + adapter from chain B synchronously handle and execute the message.

2. GasLimit only for storing a message + adapter from chain B storing the message + permissionless execute()

Could be a mess to ensure both ways always work and to decide which GasLimit will be used in the remote chain.

Reply:
BTW, to accuretely measure the `gasLimit` is `max(storing a big batch, handleARecoveryMessage)`.

Do you think we should limit the batch size? for example 2Kb or so